CMAKE.md

```diff
@@ -0,0 +1,31 @@
+# Building with CMake
+
+Some portions of this repository are portable beyond iOS and can be built using
+CMake.
+
+## Dependencies
+
+You need:
+  * `cmake`
+  * `ninja` (optional)
+
+Install with your favorite package manager for your platform, e.g. homebrew on
+macOS, chocolatey on Windows, apt-get on Ubuntu, etc.
+
+## Setup
+
+Prepare a working directory
+```
+mkdir build
+cd build
+```
+
+Configure the build
+```
+cmake -G Ninja ..
+```
+
+## Testing
+```
+ninja test
+```
```

Comment:
ninja gtest  also?

Reply:
The `gtest` target here is building the library, not actually running tests. I gave that in the review comments for the PR because it's not intended to be a primary way of interacting with this. Targets that depend upon gtest will cause it to be built.